packages/graphcache-rn-async-storage/src/makeAsyncStorage.ts

```diff
@@ -0,0 +1,106 @@
+import { StorageAdapter } from '@urql/exchange-graphcache';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import NetInfo from '@react-native-community/netinfo';
+
+type StorageOptions = {
+  dataKey?: string;
+  metadataKey?: string;
+  maxAge?: number; // Number of days
+};
+
+export const makeAsyncStorage: (ops?: StorageOptions) => StorageAdapter = ({
+  dataKey = 'graphcache-data',
+  metadataKey = 'graphcache-metadata',
+  maxAge = 7,
+} = {}) => {
+  const todayDayStamp = Math.floor(
+    new Date().valueOf() / (1000 * 60 * 60 * 24)
+  );
+  const todayBatch = {};
+  const prefixCheck = new RegExp(`^${dataKey}_(\\d+)$`);
+
+  return {
+    /**
+     * On initial read, pull storage keys and see which match our storage key pattern.
+     * Any record that is expired should be removed from AsyncStorage.
+     * The rest of the batches should be merged into a cache in chrono order.
+     * Also hydrate todayBatch if we find it.
+     */
+    readData: async () => {
+      const cache = {};
+
+      try {
+        const persistedDayStamps = (await AsyncStorage.getAllKeys())
+          .reduce<number[]>((filtered, curr) => {
+            const ts = curr.match(prefixCheck)?.[1];
```

Comment:
I don't think we currently support `?`

Reply:
Yeah, looks like tsc is happy with it, the build command not so much. Removed ✅